detection_rules/generic_loader.py

```diff
@@ -123,6 +123,10 @@ def add_item(self, item: GenericCollectionTypes) -> None:
     def load_dict(self, obj: dict, path: Optional[Path] = None) -> GenericCollectionTypes:
         """Load a dictionary into the collection."""
         is_exception = True if 'exceptions' in obj else False
+        if not RULES_CONFIG.bypass_version_lock:
+            # remove version and revision fields if bypassing version lock
```

Comment:
do we really need to also remove `revision`?

Reply:
I think so -  I think we need to treat them as tightly coupled (`version`.`revision`), meaning one cannot exist without the other (though revision is sometimes optional)

Reply:
but also - actually

version lock has no tie in to action and exceptions versions atm - meaning we should _always_ require static versions for those objects?

Reply:
and I don't think they use `revision`

Reply:
delete!